client/api/src/main/java/io/smallrye/graphql/client/core/utils/validation/NameValidation.java

```diff
@@ -0,0 +1,86 @@
+
+    /**
+     * The regular expression patterns for a valid GraphQL names.
+     */
+    private static final String NAME_PATTERN = "[a-zA-Z_][a-zA-Z0-9_]*";
+    private static final String FIELD_NAME_PATTERN = "^" + NAME_PATTERN + "(:" + NAME_PATTERN + ")?$";
+
+    /**
+     * Validates a GraphQL name and returns it. Throws an IllegalArgumentException if the name is null or invalid.
+     * Allows empty string "" as a valid input.
+     * 
+     * @param name the name to validate
+     * @return the validated name
+     * @throws IllegalArgumentException if the name is null or invalid
+     */
+    public static String setAndValidateNameAllowEmpty(String name) {
+        if (name == null || name.isEmpty()) {
+            return "";
+        } else if (!name.matches(NAME_PATTERN)) {
+            throw new IllegalArgumentException("Invalid name: " + name);
+        }
+        return name;
+    }
+
+    /**
+     * Validates a GraphQL fragment name and returns it. Throws an IllegalArgumentException if the name is null,
+     * invalid or is equal to the reserved word "on".
+     *
+     * @param name the name to validate
+     * @return the validated name
+     * @throws IllegalArgumentException if the name is null, invalid or is equal to the reserved word "on"
+     * @see <a href="https://spec.graphql.org/draft/#sec-Language.Fragments">
+     *      https://spec.graphql.org/draft/#sec-Language.Fragments
+     *      </a>
+     */
+    public static String setAndValidateFragmentName(String name) {
+        if (name == null || !name.matches(NAME_PATTERN)) {
+            throw new IllegalArgumentException("Invalid inline fragment name: " + name);
+        } else if (name.equals("on")) {
+            throw new IllegalArgumentException("Inline fragment name cannot be 'on'");
+        }
+        return name;
+    }
+
+    /**
+     * Validates a GraphQL name and returns it. Throws an IllegalArgumentException if the name is null or invalid.
+     * Does not allow empty string "" as a valid input.
+     * 
+     * @param name the name to validate
+     * @return the validated name
+     * @throws IllegalArgumentException if the name is null, invalid or empty
+     */
+    public static String setAndValidateName(String name) {
+        if (name == null || !name.matches(NAME_PATTERN)) {
+            throw new IllegalArgumentException("Invalid name: " + name);
+        }
+        return name;
+    }
+
+    /**
+     * Validates the given field name, allowing at most one colon that is not at the beginning or end
+     * for alias.
+     *
+     * @param fieldName the field name to validate
+     * @return the validated field name
+     * @throws IllegalArgumentException if the field name is null or invalid
+     * @see <a href="https://spec.graphql.org/draft/#sec-Language.Fields">
+     *      https://spec.graphql.org/draft/#sec-Language.Fields
+     *      </a>
+     */
+    public static String setAndValidateFieldName(String fieldName) {
```

Comment:
why `setAndValidateFieldName`? You're not setting it anywhere here (the same applies to other methods in this class)